mapApp/static/mapApp/js/index.js

```diff
@@ -453,36 +465,38 @@ function getXHRPopup(layer) {
         .setContent("Loading data off server ...")
         .openOn(map);
 
+    // PK is stored in under key 'pk' for data loaded from database, 'id' for points just created and added to incidentData from submitted form
+    let pk = feature.properties.pk || feature.properties.id;
+
     if (type === "newInfrastructure") {
```

Comment:
I find this if/else block a little confusing (line 471), like I want to know why one the if path adds 's_xhr' and the else path only adds 'xhr'. I think it's because we use 'newInfrastructure' for the type here client side, but the matching endpoint name/URL pattern is 'newInfrastructures'.

What do you think about doing something like this:

// Correct mismatch between client side type and required Django URL pattern for newInfrastructures endpoint.
if (type === "newInfrastructure") {
  type = type + 's';
}

loadPopupDetails(pk, popup, type, "//" + hostname + "/" + type + "_xhr?format=json&pk=");

Alternatively, we could just add a couple new URL patterns to mapAPp/urls.py to like this:

```
	url(r'^newInfrastructure_tiny/?$', views.TinyNewInfrastructureList.as_view(), name='tiny-newInfrastructures-list'),
	url(r'^newInfrastructure_xhr/?$', views.XHRNewInfrastructureInfo.as_view(), name='xhr-newInfrastructures-detail'),
```
Then we wouldn't need any if/else statements.

Reply:
I couldn't find any reason why newInfrastructures had a different URL pattern that the other datasets, removing the 's' didn't break retrieving the XHR data and there were no other references to it in the codebase... is there something I'm missing or is this just the simplest/best fix? commit 5fc902